### ISSUE TITLE
Make right-clicking ports (on both AND/OR nodes and interfaces) easier

### ISSUE THREAD

**Issue by danionita** [danionita/e3tools#27]
Currently, selecting ports (and not edges) is very tricky)

**Comment by bobismijnnaam** [comment]
How would we go about this? An easy solution would be to make them even bigger, but that might make the screen more cluttered. We could also do some snapping of the mouse to prevent clicking something else when you're close to the signal ports, but I'm not sure if JGraphX really supports that. We could also make all the signal port properties part of one of the sides of the value exchange, s.t. when you right click on the right side of an edge you can also change all the signal port properties. This does mean the port needs to be connected though

**Comment by danionita** [comment]
The problem right now is that when trying to click a port, you sometimes
end up selecting the edge the is connected to it. This problem occurs on
AND/OR gate ports too.

I like both the snapping and the enlargement solution. Maybe a little bit
of both? Or if snapping is not straight-forward, I would stick to a couple
of points size increase.

**Comment by bobismijnnaam** [comment]
I understand the problem for and/or gates. I think the snapping would work well there. But why would the user right click on the signal dot of a value port? At the moment it does not even have a context menu. Should it?

**Comment by bobismijnnaam** [comment]
I implemented very basic and conservate right-click snapping towards logic dots (black dots on logic elements) in 835f631f49567e3d5770713ae65af6b7913c4ec4. It's a tricky feature though so we should evaluate it thoroughly.

**Comment by danionita** [comment]
You're right, the user would never right click the black dot. The issue was
about the ports.

**Comment by danionita** [comment]
Playing around with the tool, I still have trouble right-clicking value-ports. The behavior is more confusing and more unpredictable than even the old tool. 

I think this is mainly due to the "Grabbing" point on the end of each edge. I am referring to the small red square I assume was supposed to be used to re-attach the edge to something else. When hovering over it, the mouse becomes a "hand". That adds to the confusion, since a hand also appears when hovering over the center of the port and one can never which one he is going to click. Having that red square does not make sense anyway since transfers cannot apparently be re-attached?!

Furthermore, there seems to be a difference between the "connection area" of a port (center of the triangle) and the "selection area" (currently an invisible square wrapping the triangle). To make matters worse, a green square is displayed when hovering over the "connection area", but not the "selection area" which actually corresponds to that square. Having different areas does not make sense since - unlike other elements -  the user cannot move ports.

Therefore, can we make it so that:
- the square (yellow highlight in the image below) around a ValuePort  acts as both a "selection" and a "connection area". That is, when either left- or right-clicked it becomes "selected" by displaying the green square (just like any other element). In addition, keeping the left-button pressed allows connecting it to other ports and a right-click opens the context menu. The "hand" cursor is only displayed when hovering over this area. 
- the area of a Value Interface which does not correspond to a port or a node  (red highlighting in the photo) should select the value interface. That means it should impossible to select a Value Exchange when clicking anywhere inside a Value Interface.
- the red square appears outside the Value Interface and actually works. As in, you can re-attach value transfers like in the old tool.

![image](https://cloud.githubusercontent.com/assets/11060388/19355634/57f40dda-916b-11e6-9bbb-4b5890a9c0a6.png)

**Comment by bobismijnnaam** [comment]
Removed the edge handles (red boxes at the end of connection elements/value exchanges). All the normal edge features should still work as they did before. Does not improve selectability a lot but makes it look better at least.

**Comment by bobismijnnaam** [comment]
As of 8351bfef5861ea3cbb539900b47a7d143ce0a834.

**Comment by danionita** [comment]
8351bfe seems to have only addressed my first comment (about the confusing "grabbing point"/:edge handle:) and not my second comment (about the green square only being shown when hovering over the center of the triangle). 
Furthermore, **edges can now no longer be selected**. This means (1) they cannot be deleted, (2) their labels cannot be moved.

Could you:
- increase the size of the area where the green square is shown so that it is closer in size to the actual size of the area which corresponds to the port?
- Make edges selectable again?

**Comment by danionita** [comment]
I've made edges selectable again. But we should still find a way to increase the size of the area where the green square is shown so that it is closer in size to the actual size of the area which corresponds to the port.
